BrowserLauncher error

It seems it is not getting the variables and it is giving a new error:

Run pnpm start
  pnpm start
  shell: /usr/bin/bash -e {0}
  env:
    PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
    EMAIL: 
    PASSWORD: 
    BASE_URL: 
    TWO_CAPTCHA_API_KEY: 
    MONDAY: 
    TUESDAY: 
    WEDNESDAY: 
    THURSDAY: 
    FRIDAY: 
    SATURDAY: 
    SUNDAY: 
    AVAILABLE_DAYS: 

> autowod@1.0.0 start /home/runner/work/aut/aut
> tsx src/index.ts

/home/runner/work/aut/aut/node_modules/.pnpm/puppeteer-core@[2](https://github.com/diskol0/aut/actions/runs/13955918008/job/39066854889#step:6:2)3.11.1/node_modules/puppeteer-core/src/node/BrowserLauncher.ts:470
          throw new Error(
                ^


Error: Could not find Chrome (ver. 1[3](https://github.com/diskol0/aut/actions/runs/13955918008/job/39066854889#step:6:3)1.0.6778.204). This can occur if either
 1. you did not perform an installation before running the script (e.g. `npx puppeteer browsers install chrome`) or
 2. your cache path is incorrectly configured (which is: /home/runner/.cache/puppeteer).
For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.
    at ChromeLauncher.resolveExecutablePath (/home/runner/work/aut/aut/node_modules/.pnpm/puppeteer-core@23.11.1/node_modules/puppeteer-core/src/node/BrowserLauncher.ts:[4](https://github.com/diskol0/aut/actions/runs/13955918008/job/39066854889#step:6:4)70:17)
    at ChromeLauncher.computeLaunchArguments (/home/runner/work/aut/aut/node_modules/.pnpm/puppeteer-core@23.11.1/node_modules/puppeteer-core/src/node/ChromeLauncher.ts:128:1[6](https://github.com/diskol0/aut/actions/runs/13955918008/job/39066854889#step:6:6))
    at async ChromeLauncher.launch (/home/runner/work/aut/aut/node_modules/.pnpm/puppeteer-core@23.11.1/node_modules/puppeteer-core/src/node/BrowserLauncher.ts:9[7](https://github.com/diskol0/aut/actions/runs/13955918008/job/39066854889#step:6:7):24)
    at async main (/home/runner/work/aut/aut/src/index.ts:14:1[9](https://github.com/diskol0/aut/actions/runs/13955918008/job/39066854889#step:6:9))

Node.js v20.19.0
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.

--- a/.github/workflows/daily-reservation.yml
+++ b/.github/workflows/daily-reservation.yml
@@ -1,55 +1,47 @@
-# This workflow runs every day to book your workout sessions
 name: Daily Reservation
+run-name: "🏋️ AutoWOD – ${{ github.event_name == 'workflow_dispatch' && 'Manual run' || 'Daily schedule' }}"
 
-# When does this run?
 on:
   schedule:
-    # We run twice to roughly hit 22:30 in Spain all year:
-    # - 20:31 UTC:
-    #     * 21:31 in winter (CET, UTC+1)
-    #     * 22:31 in summer (CEST, UTC+2)
-    # - 21:31 UTC:
-    #     * 22:31 in winter (CET, UTC+1)
-    #     * 23:31 in summer (CEST, UTC+2)
-    # This guarantees at least one run around 22:30 local time in both seasons,
-    # at the cost of one extra daily run.
+    # Runs twice to cover both Spain summer (CEST, UTC+2) and winter (CET, UTC+1) time.
+    # One of these will always land around 22:30 local time.
     - cron: '31 20 * * *'
     - cron: '31 21 * * *'
 
-  # Also allows you to run it manually from the Actions tab.
-  # Useful for gyms that open the full week on a single day — trigger once
-  # and set available_days to the number of days you want to book ahead.
+  # Run it manually from the Actions tab whenever you need to.
+  # Useful when your gym opens the whole week at once — trigger once and
+  # set "Days to book ahead" to the number of days you want to cover.
   workflow_dispatch:
     inputs:
       available_days:
-        description: 'Days to book ahead (leave empty to use AVAILABLE_DAYS secret)'
+        description: 'Days to book ahead (leave empty to use the AVAILABLE_DAYS variable)'
         required: false
         default: ''
 
-# The tasks to run
+# Prevent two runs from overlapping and trying to book the same sessions.
+concurrency:
+  group: reservation
+  cancel-in-progress: false
+
 jobs:
   book-session:
     name: Book Workout Session
     runs-on: ubuntu-latest
 
     steps:
-      # Get a copy of your code
       - name: 📥 Get latest code
         uses: actions/checkout@v4
 
-      # Set up Node.js environment
       - name: 🔧 Set up Node.js
         uses: actions/setup-node@v4
         with:
           node-version: '20'
 
-      # Install pnpm (package manager)
       - name: 📦 Install pnpm
         uses: pnpm/action-setup@v4
         with:
           version: latest
 
-      # Cache pnpm store to speed up installs
       - name: 💾 Cache pnpm store
         uses: actions/cache@v4
         with:
@@ -58,11 +50,9 @@ jobs:
           restore-keys: |
             pnpm-store-${{ runner.os }}-
 
-      # Install project dependencies
       - name: 📚 Install dependencies
         run: pnpm install --frozen-lockfile
 
-      # Cache installed Chrome for Puppeteer
       - name: 💾 Cache Puppeteer browser
         uses: actions/cache@v4
         with:
@@ -71,32 +61,23 @@ jobs:
           restore-keys: |
             puppeteer-cache-${{ runner.os }}-
 
-      # Ensure Chrome for Puppeteer is installed (uses cache when available)
       - name: 🌐 Install Chrome for Puppeteer
         run: pnpm exec puppeteer browsers install chrome
 
-      # Run the booking script
       - name: 🎯 Run booking script
         env:
-          # Authentication
+          # Secrets (sensitive — set in Settings → Secrets and variables → Actions → Secrets tab)
           EMAIL: ${{ secrets.EMAIL }}
           PASSWORD: ${{ secrets.PASSWORD }}
-
-          # Target Website
-          BASE_URL: ${{ secrets.BASE_URL }}
-
-          # 2Captcha API Key
           TWO_CAPTCHA_API_KEY: ${{ secrets.TWO_CAPTCHA_API_KEY }}
 
-          # Schedule
-          MONDAY: ${{ secrets.MONDAY }}
-          TUESDAY: ${{ secrets.TUESDAY }}
-          WEDNESDAY: ${{ secrets.WEDNESDAY }}
-          THURSDAY: ${{ secrets.THURSDAY }}
-          FRIDAY: ${{ secrets.FRIDAY }}
-          SATURDAY: ${{ secrets.SATURDAY }}
-          SUNDAY: ${{ secrets.SUNDAY }}
-
-          # Configuration (manual trigger input takes precedence over secret)
-          AVAILABLE_DAYS: ${{ inputs.available_days || secrets.AVAILABLE_DAYS }}
+          # Variables (set in Settings → Secrets and variables → Actions → Variables tab)
+          MONDAY: ${{ vars.MONDAY }}
+          TUESDAY: ${{ vars.TUESDAY }}
+          WEDNESDAY: ${{ vars.WEDNESDAY }}
+          THURSDAY: ${{ vars.THURSDAY }}
+          FRIDAY: ${{ vars.FRIDAY }}
+          SATURDAY: ${{ vars.SATURDAY }}
+          SUNDAY: ${{ vars.SUNDAY }}
+          AVAILABLE_DAYS: ${{ inputs.available_days || vars.AVAILABLE_DAYS }}
         run: pnpm start

--- a/.github/workflows/daily-reservation.yml
+++ b/.github/workflows/daily-reservation.yml
@@ -16,8 +16,15 @@ on:
     - cron: '31 20 * * *'
     - cron: '31 21 * * *'
 
-  # Also allows you to run it manually from the Actions tab
+  # Also allows you to run it manually from the Actions tab.
+  # Useful for gyms that open the full week on a single day — trigger once
+  # and set available_days to the number of days you want to book ahead.
   workflow_dispatch:
+    inputs:
+      available_days:
+        description: 'Days to book ahead (leave empty to use AVAILABLE_DAYS secret)'
+        required: false
+        default: ''
 
 # The tasks to run
 jobs:
@@ -90,6 +97,6 @@ jobs:
           SATURDAY: ${{ secrets.SATURDAY }}
           SUNDAY: ${{ secrets.SUNDAY }}
 
-          # Configuration
-          AVAILABLE_DAYS: ${{ secrets.AVAILABLE_DAYS }}
+          # Configuration (manual trigger input takes precedence over secret)
+          AVAILABLE_DAYS: ${{ inputs.available_days || secrets.AVAILABLE_DAYS }}
         run: pnpm start

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -1,0 +1,49 @@
+# CLAUDE.md
+
+This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.
+
+## Commands
+
+```bash
+pnpm dev          # Run bot locally (loads .env automatically)
+pnpm start        # Run bot in production mode
+pnpm typecheck    # TypeScript type checking
+pnpm lint         # Lint TypeScript files
+pnpm format       # Format with Prettier
+```
+
+No test suite exists yet — `pnpm test` is a no-op placeholder.
+
+## Architecture
+
+AutoWOD is a headless browser bot that automatically books CrossFit sessions on WODBuster-powered gym platforms. It runs on a cron schedule via GitHub Actions.
+
+**Main flow (`src/index.ts`):**
+1. Launch Puppeteer (headless in CI, headed locally — detected via `CI` env var)
+2. Solve Cloudflare Turnstile CAPTCHA via 2Captcha API (CI only)
+3. Authenticate with email/password (optional 2FA device selection)
+4. Navigate the reservations page and book sessions for each upcoming day
+5. Log a summary of results (booked / waitlisted / already booked / skipped)
+
+**Services (`src/services/`):**
+- `auth.ts` — login form submission and 2FA handling
+- `captcha.ts` — injects `scripts/captcha-interceptor.js` into the page to intercept Turnstile parameters, sends them to 2Captcha, then injects the token back
+- `reservation.ts` — iterates over available days, reads button state labels (Spanish: "Entrenar" = available, "Avisar" = waitlist, "Borrar" = already booked), and performs bookings
+
+**Configuration (`src/config.ts`):**
+All config is loaded from environment variables. The `.env.example` file lists every variable. Key ones:
+
+| Variable | Purpose |
+|---|---|
+| `EMAIL`, `PASSWORD` | Gym account credentials |
+| `TWO_CAPTCHA_API_KEY` | 2Captcha API key for CAPTCHA solving |
+| `BASE_URL` | Gym's WODBuster URL |
+| `MONDAY`–`SUNDAY` | Preferred time slot per day (24h format, e.g. `17:00`) |
+| `AVAILABLE_DAYS` | Booking horizon in days (default: 7) |
+| `CI` | Set by GitHub Actions; enables headless mode and CAPTCHA solving |
+
+Copy `.env.example` to `.env` for local development.
+
+## GitHub Actions
+
+The workflow in `.github/workflows/` runs the bot on a cron schedule. Secrets map directly to the environment variables above. The `CI=true` flag is what switches the bot from local (headed browser, no CAPTCHA) to cloud (headless, CAPTCHA solving enabled) mode.

--- a/README.md
+++ b/README.md
@@ -1,169 +1,157 @@
 # AutoWOD
 
-Automatically book your workout sessions using GitHub Actions! No coding knowledge required. This tool runs in the cloud and handles the booking process for you.
+[![Daily Reservation](https://github.com/RubenGlez/autowod/actions/workflows/daily-reservation.yml/badge.svg)](https://github.com/RubenGlez/autowod/actions/workflows/daily-reservation.yml)
 
-> **Important**: This tool is specifically designed for gyms using the [WODBuster](https://wodbuster.com/) platform. It will not work with other booking systems.
+Automatically books your workout sessions on [WODBuster](https://wodbuster.com/) — the booking platform used by many CrossFit gyms. Once set up, it runs every day in the cloud so you never have to worry about grabbing a spot.
 
-## What does this do?
+> **Is your gym on WODBuster?** Check your gym's booking page URL. If it contains `wodbuster.com`, you're good to go.
 
-- ✨ Automatically books your workout sessions on WODBuster
-- 🔄 Runs daily in the cloud (using GitHub Actions)
-- 🔒 Handles login and verification automatically
-- ⏰ Books sessions at your preferred times
-- 🆓 Completely free to use
+---
 
-## Before You Start
+## What you need before starting
 
-1. **Check Your Gym**: Make sure your gym uses WODBuster as their booking platform. You can verify this by checking if your gym's booking URL contains `wodbuster.com`.
+- A GitHub account (free) — [sign up here](https://github.com/join) if you don't have one
+- Your WODBuster login email and password
+- A 2Captcha API key (costs less than $1/month) — [see step 4 below](#4-get-a-2captcha-api-key)
 
-2. **Terms of Service**: Please ensure you have permission from your gym to automate bookings through their WODBuster platform.
+---
 
-## Quick Start Guide (No Coding Required!)
+## Setup (one time only)
 
-1. **Fork this repository**
+### 1. Fork this repository
 
-   - Click the "Fork" button at the top right of this page
-   - This creates your own copy of the tool
+Click the **Fork** button at the top right of this page. This creates your own private copy of the tool under your GitHub account.
 
-2. **Set up your secrets**
+### 2. Add your secrets
 
-   - Go to your forked repository's Settings
-   - Click on "Secrets and variables" → "Actions"
-   - Add these required secrets by clicking "New repository secret":
+Secrets are for sensitive information like your password. GitHub keeps them encrypted and never shows them in logs.
 
-     ```
-     # Authentication
-     EMAIL=your-email@example.com
-     PASSWORD=your-password
+1. In your forked repository, click **Settings**
+2. In the left sidebar, click **Secrets and variables → Actions**
+3. Make sure you're on the **Secrets** tab
+4. Click **New repository secret** and add each of the following:
 
-     # 2Captcha API Key
-     TWO_CAPTCHA_API_KEY=your-2captcha-api-key
+| Name | Value |
+|------|-------|
+| `EMAIL` | Your WODBuster login email |
+| `PASSWORD` | Your WODBuster password |
+| `TWO_CAPTCHA_API_KEY` | Your 2Captcha API key (see step 4) |
 
-     # Target Website
-     BASE_URL=https://your-gym-website.com
-     ```
+### 3. Set your weekly schedule
 
-3. **Set your workout schedule**
+Your schedule uses **Variables** — unlike secrets, you can see and edit them at any time without deleting and recreating them.
 
-   - Still in the repository secrets, add your preferred times for each day:
-     ```
-     # Schedule (use 24-hour format)
-     MONDAY=17:00
-     TUESDAY=18:30
-     WEDNESDAY=09:00
-     THURSDAY=19:30
-     FRIDAY=20:15
-     SATURDAY=14:30
-     SUNDAY=14:30
-     ```
-   - For days you don't want to book, either:
-     - Don't add the day's secret at all, or
-     - Set it to an empty value
-   - Use 24-hour format (e.g., '14:30' for 2:30 PM)
+1. Still in **Settings → Secrets and variables → Actions**
+2. Click the **Variables** tab
+3. Click **New repository variable** and add a variable for each day you want to book
 
-   **Multiple classes at the same time?** If your gym offers more than one class at the same start time (e.g. CrossFit and Endurance both at 18:00), add the class name after a `|`:
-     ```
-     MONDAY=18:00|CrossFit
-     TUESDAY=18:00|Endurance
-     ```
-   The name is matched case-insensitively. If no match is found it falls back to the first available slot at that time.
+| Name | Value example | Meaning |
+|------|--------------|---------|
+| `MONDAY` | `18:00` | Book the 6 PM class every Monday |
+| `TUESDAY` | `18:00` | Book the 6 PM class every Tuesday |
+| `WEDNESDAY` | `09:00` | Book the 9 AM class every Wednesday |
+| `THURSDAY` | `18:00` | |
+| `FRIDAY` | `18:00` | |
+| `SATURDAY` | `10:00` | |
+| `SUNDAY` | `10:00` | |
 
-4. **Optional: Configure booking window**
+- Use 24-hour format (`18:00` not `6:00 PM`)
+- Only add variables for days you actually want to book — skip the rest
 
-   ```
-   # How many days in advance you can book (default: 7)
-   AVAILABLE_DAYS=7
-   ```
+**Two classes at the same time?** If your gym offers multiple classes at the same start time (e.g. CrossFit and Endurance both at 18:00), add the class name after a `|`:
 
-   **Gym opens the whole week at once?** If your gym releases all reservations on a single day (e.g. every Sunday), you don't need to run the script daily. Just trigger it manually from the **Actions** tab once your gym opens slots:
-   1. Go to **Actions → Daily Reservation → Run workflow**
-   2. Set *Days to book ahead* to however many days you want (e.g. `7`)
-   3. Click **Run workflow**
+```
+MONDAY=18:00|CrossFit
+TUESDAY=18:00|Endurance
+```
 
-   This books all the days you've configured in one shot.
+**Optional — how many days ahead to book:**
 
-5. **Get a 2Captcha API Key**
+| Name | Value | Meaning |
+|------|-------|---------|
+| `AVAILABLE_DAYS` | `7` | Book up to 7 days in advance (default if not set) |
 
-   - Go to [2captcha.com](https://2captcha.com)
-   - Register an account
-   - Add funds (minimum $3)
-   - Copy your API key from your account dashboard
+### 4. Get a 2Captcha API key
 
-6. **Enable GitHub Actions**
-   - Go to the "Actions" tab in your repository
-   - Click "I understand my workflows, go ahead and enable them"
+The tool needs this to handle the login security check automatically.
 
-That's it! The tool will now automatically try to book your preferred sessions every day at 22:30 (winter) or 21:30 (summer) Spain time.
+1. Go to [2captcha.com](https://2captcha.com) and create a free account
+2. Add a small amount of credit (minimum ~$3, lasts months at ~$0.001 per run)
+3. Copy your API key from the dashboard and add it as the `TWO_CAPTCHA_API_KEY` secret
 
-## Checking if it's Working
+### 5. Enable GitHub Actions
 
-1. Go to the "Actions" tab in your repository
-2. Look for the latest "Daily Reservation" workflow run
-3. Click on it to see the details and logs
-4. A green checkmark means it ran successfully
+1. Click the **Actions** tab in your forked repository
+2. If prompted, click **I understand my workflows, go ahead and enable them**
+
+That's it! The tool will now run automatically every evening around 22:30 Spain time.
+
+---
+
+## Checking if it worked
+
+1. Click the **Actions** tab in your repository
+2. Click the latest **AutoWOD** run
+3. You'll see a summary card showing exactly what happened for each day — no need to read through logs
+
+A green checkmark means the run completed. Check the summary card to see which sessions were booked.
+
+---
+
+## Changing your schedule
+
+Just go to **Settings → Secrets and variables → Actions → Variables tab**, click the variable you want to change (e.g. `MONDAY`), and update the value. The next run will pick it up automatically.
+
+---
+
+## My gym opens all reservations on one day
+
+Some gyms release the full week's schedule at once (e.g. every Sunday). Instead of running daily, you can trigger the tool manually:
+
+1. Go to **Actions → Daily Reservation**
+2. Click **Run workflow** (top right)
+3. In the *Days to book ahead* field, enter the number of days you want to cover (e.g. `7`)
+4. Click **Run workflow**
+
+This books everything in one go.
+
+---
 
 ## Troubleshooting
 
-### Common Issues
+**The run shows a red ✗**
+- Open the run and check the summary card or the step logs for the error message
+- Most common cause: a secret is missing or has a typo — double-check `EMAIL` and `PASSWORD`
 
-1. **Workflow not running?**
+**All days show "Skipped" in the summary**
+- You haven't added any day variables yet — go to the Variables tab and add at least one (e.g. `MONDAY=18:00`)
 
-   - Make sure you've enabled GitHub Actions
-   - Check if all required secrets are properly set
+**"Could not find Chrome" error**
+- Make sure your fork is up to date with the latest version of this repository
+- Do not remove or modify the *Install Chrome for Puppeteer* step in the workflow file
 
-2. **Login fails?**
+**Wrong class booked when multiple classes share a time**
+- Add the class name to your variable: `MONDAY=18:00|CrossFit`
+- The name match is not case-sensitive
 
-   - Double-check your EMAIL and PASSWORD secrets
-   - Make sure your BASE_URL is correct
+**Captcha errors**
+- Check that your `TWO_CAPTCHA_API_KEY` secret is correct
+- Make sure your 2Captcha account has credit remaining
 
-3. **Captcha issues?**
+---
 
-   - Ensure your TWO_CAPTCHA_API_KEY is correct
-   - Check if you have sufficient balance on 2captcha.com
+## Privacy & cost
 
-4. **Wrong booking times?**
-   - Make sure you're using 24-hour format (e.g., "17:00" not "5:00 PM")
-   - Check if the day's variable is set correctly (e.g., "MONDAY=17:00")
+- Your email, password, and API key are stored as encrypted GitHub secrets — only your workflow can read them
+- GitHub Actions is free for public repositories
+- 2Captcha costs roughly $0.001 per run — less than $1/month for daily bookings
 
-5. **Could not find Chrome error?**
-   - Make sure you are using `puppeteer` (not `puppeteer-core`) in your `package.json`
-   - The workflow installs Chrome automatically — do not remove the *Install Chrome for Puppeteer* step
-   - If you forked this repo a long time ago, update your workflow file from the latest version of this repo
-
-6. **Script runs but skips all days?**
-   - Check that your day secrets are set and not empty (e.g. `MONDAY=18:00`)
-   - The script now logs a warning at startup if no days are configured
-
-### Need Help?
-
-If you're having problems:
-
-1. Check the "Issues" tab to see if others had the same problem
-2. Create a new issue describing your problem
-3. Include the workflow run logs (without any personal information)
-
-## Cost Considerations
-
-- GitHub Actions: Free for public repositories
-- 2Captcha: Approximately $0.001 per booking attempt
-- Estimated monthly cost: Less than $1 for daily bookings
-
-## Safety & Privacy
-
-- ✅ Your credentials are securely stored as GitHub secrets
-- ✅ No one can see your email, password, or API keys
-- ✅ The tool only accesses your gym's booking system
-- ❌ Never share your secrets or API keys with anyone
-
-## License
-
-This project is licensed under the ISC License - see the LICENSE file for details.
+---
 
 ## Disclaimer
 
-This tool is specifically designed for WODBuster-powered gyms and is provided for educational purposes only. Please ensure you:
+This tool is designed for WODBuster-powered gyms. Please make sure you have permission from your gym to automate bookings, and use it responsibly in line with their booking policies.
 
-1. Have confirmed your gym uses the WODBuster platform
-2. Have permission to automate interactions with your gym's WODBuster website
-3. Comply with both WODBuster's and your gym's terms of service
-4. Use this tool responsibly and in accordance with your gym's booking policies
+## License
+
+ISC — see the [LICENSE](LICENSE) file for details.

--- a/README.md
+++ b/README.md
@@ -61,12 +61,26 @@ Automatically book your workout sessions using GitHub Actions! No coding knowled
      - Set it to an empty value
    - Use 24-hour format (e.g., '14:30' for 2:30 PM)
 
+   **Multiple classes at the same time?** If your gym offers more than one class at the same start time (e.g. CrossFit and Endurance both at 18:00), add the class name after a `|`:
+     ```
+     MONDAY=18:00|CrossFit
+     TUESDAY=18:00|Endurance
+     ```
+   The name is matched case-insensitively. If no match is found it falls back to the first available slot at that time.
+
 4. **Optional: Configure booking window**
 
    ```
    # How many days in advance you can book (default: 7)
    AVAILABLE_DAYS=7
    ```
+
+   **Gym opens the whole week at once?** If your gym releases all reservations on a single day (e.g. every Sunday), you don't need to run the script daily. Just trigger it manually from the **Actions** tab once your gym opens slots:
+   1. Go to **Actions → Daily Reservation → Run workflow**
+   2. Set *Days to book ahead* to however many days you want (e.g. `7`)
+   3. Click **Run workflow**
+
+   This books all the days you've configured in one shot.
 
 5. **Get a 2Captcha API Key**
 
@@ -110,6 +124,15 @@ That's it! The tool will now automatically try to book your preferred sessions e
 4. **Wrong booking times?**
    - Make sure you're using 24-hour format (e.g., "17:00" not "5:00 PM")
    - Check if the day's variable is set correctly (e.g., "MONDAY=17:00")
+
+5. **Could not find Chrome error?**
+   - Make sure you are using `puppeteer` (not `puppeteer-core`) in your `package.json`
+   - The workflow installs Chrome automatically — do not remove the *Install Chrome for Puppeteer* step
+   - If you forked this repo a long time ago, update your workflow file from the latest version of this repo
+
+6. **Script runs but skips all days?**
+   - Check that your day secrets are set and not empty (e.g. `MONDAY=18:00`)
+   - The script now logs a warning at startup if no days are configured
 
 ### Need Help?
 

--- a/src/config.ts
+++ b/src/config.ts
@@ -1,6 +1,6 @@
 import { ReservationPreferences } from './types';
 
-export const baseUrl = process.env.BASE_URL || 'https://wodbuster.com';
+export const baseUrl = 'https://wodbuster.com';
 
 export const email = process.env.EMAIL ?? '';
 export const password = process.env.PASSWORD ?? '';

--- a/src/index.ts
+++ b/src/index.ts
@@ -14,12 +14,11 @@ function validateConfig() {
   const missing: string[] = [];
   if (!email) missing.push('EMAIL');
   if (!password) missing.push('PASSWORD');
-  if (!baseUrl) missing.push('BASE_URL');
 
   if (missing.length > 0) {
     console.error(
-      `❌ Missing required environment variables: ${missing.join(', ')}\n` +
-        `Make sure these are set as GitHub repository secrets (Settings → Secrets and variables → Actions).`
+      `❌ Missing required secrets: ${missing.join(', ')}\n` +
+        `Go to your repository Settings → Secrets and variables → Actions and add them.`
     );
     process.exit(1);
   }
@@ -27,8 +26,8 @@ function validateConfig() {
   const hasAnyDay = Object.values(reservationsPreferences).some(v => v);
   if (!hasAnyDay) {
     console.warn(
-      `⚠️ No days configured — set at least one of MONDAY…SUNDAY to a time (e.g. MONDAY=18:00). ` +
-        `The script will run but skip all days.`
+      `⚠️ No days configured — set at least one of MONDAY…SUNDAY in your repository variables ` +
+        `(Settings → Secrets and variables → Actions → Variables tab).`
     );
   }
 }

--- a/src/index.ts
+++ b/src/index.ts
@@ -10,7 +10,31 @@ import {
   reservationsPreferences,
 } from './config';
 
+function validateConfig() {
+  const missing: string[] = [];
+  if (!email) missing.push('EMAIL');
+  if (!password) missing.push('PASSWORD');
+  if (!baseUrl) missing.push('BASE_URL');
+
+  if (missing.length > 0) {
+    console.error(
+      `❌ Missing required environment variables: ${missing.join(', ')}\n` +
+        `Make sure these are set as GitHub repository secrets (Settings → Secrets and variables → Actions).`
+    );
+    process.exit(1);
+  }
+
+  const hasAnyDay = Object.values(reservationsPreferences).some(v => v);
+  if (!hasAnyDay) {
+    console.warn(
+      `⚠️ No days configured — set at least one of MONDAY…SUNDAY to a time (e.g. MONDAY=18:00). ` +
+        `The script will run but skip all days.`
+    );
+  }
+}
+
 async function main() {
+  validateConfig();
   const browser = await launch({
     headless: isCI,
     slowMo: isCI ? 0 : 50,

--- a/src/services/reservation.ts
+++ b/src/services/reservation.ts
@@ -7,6 +7,15 @@ import {
 } from '../types';
 import { availableDays } from '../config';
 
+export function parsePreferenceValue(value: string | null): {
+  time: string | null;
+  className: string | null;
+} {
+  if (!value) return { time: null, className: null };
+  const [time, className] = value.split('|');
+  return { time: time.trim() || null, className: className?.trim() || null };
+}
+
 export async function goToReservations(page: Page): Promise<void> {
   const today = new Date();
   today.setUTCHours(0, 0, 0, 0);
@@ -19,7 +28,7 @@ export async function goToReservations(page: Page): Promise<void> {
 }
 
 export async function getReservationState(
-  reservationButton: ElementHandle<HTMLButtonElement>
+  reservationButton: ElementHandle<Element>
 ): Promise<ButtonText | null> {
   const buttonText = await reservationButton.evaluate(el => el.textContent);
   return buttonText as ButtonText | null;
@@ -40,7 +49,7 @@ export async function getWeekDayFromUrl(page: Page): Promise<string> {
   const weekDayInSeconds = url.split('=')[1];
   const weekDay = new Date(Number(weekDayInSeconds) * 1000);
   return weekDay
-    .toLocaleDateString('en-US', { weekday: 'long' })
+    .toLocaleDateString('en-US', { weekday: 'long', timeZone: 'UTC' })
     .toLocaleLowerCase();
 }
 
@@ -54,10 +63,39 @@ export async function getDateFromUrl(page: Page): Promise<string> {
   }).format(new Date(Number(weekDayInSeconds) * 1000));
 }
 
+async function findReservationButton(
+  page: Page,
+  reservationKey: string,
+  className: string | null
+): Promise<ElementHandle<Element> | null> {
+  const buttons = await page.$$(
+    `div[data-magellan-destination="${reservationKey}"] button`
+  );
+
+  if (buttons.length === 0) return null;
+  if (!className || buttons.length === 1) return buttons[0];
+
+  for (const button of buttons) {
+    const sectionText = await button.evaluate(el => {
+      const section = el.closest('[data-magellan-destination]');
+      return section?.textContent ?? '';
+    });
+    if (sectionText.toLowerCase().includes(className.toLowerCase())) {
+      return button;
+    }
+  }
+
+  console.log(
+    `⚠️ Class "${className}" not found at this time slot — using first available`
+  );
+  return buttons[0];
+}
+
 export async function makeReservation(
   page: Page,
-  time: string | null
+  preference: string | null
 ): Promise<ReservationResult> {
+  const { time, className } = parsePreferenceValue(preference);
   const weekDay = await getWeekDayFromUrl(page);
   const pageTitle = await page.$('.mainTitle');
   const pageTitleText = (await pageTitle?.evaluate(el => el.textContent)) ?? '';
@@ -71,8 +109,10 @@ export async function makeReservation(
   }
 
   const reservationKey = getReservationKey(time);
-  const reservationButton = await page.$(
-    `div[data-magellan-destination="${reservationKey}"] button`
+  const reservationButton = await findReservationButton(
+    page,
+    reservationKey,
+    className
   );
 
   if (!reservationButton) {

--- a/src/services/reservation.ts
+++ b/src/services/reservation.ts
@@ -1,3 +1,4 @@
+import { appendFileSync } from 'fs';
 import { Page, ElementHandle } from 'puppeteer';
 import {
   ButtonText,
@@ -175,10 +176,64 @@ export async function makeReservation(
   return result;
 }
 
+function writeJobSummary(
+  dayResults: Array<{ weekDay: string; result: ReservationResult }>,
+  counts: {
+    booked: number;
+    waitlisted: number;
+    alreadyBooked: number;
+    skipped: number;
+    other: number;
+  }
+) {
+  const summaryFile = process.env.GITHUB_STEP_SUMMARY;
+  if (!summaryFile) return;
+
+  const statusLabel = (result: ReservationResult): string => {
+    if (!result.time) return '⏭️ Skipped';
+    if (result.state === 'Entrenar' && result.success) return '✅ Booked';
+    if (result.state === 'Avisar' && result.success) return '⏳ Waitlisted';
+    if (result.state === 'Borrar') return 'ℹ️ Already booked';
+    if (result.state === 'Finalizada') return '❌ Class already finished';
+    if (result.state === 'Cambiar') return '⚠️ Booked at a different time';
+    return '🔍 Slot not found';
+  };
+
+  const rows = dayResults.map(({ weekDay, result }) => {
+    const day = weekDay.charAt(0).toUpperCase() + weekDay.slice(1);
+    const time = result.time ?? '—';
+    return `| ${day} | ${time} | ${statusLabel(result)} |`;
+  });
+
+  const totals = [
+    counts.booked > 0 ? `**${counts.booked} booked**` : null,
+    counts.waitlisted > 0 ? `${counts.waitlisted} waitlisted` : null,
+    counts.alreadyBooked > 0 ? `${counts.alreadyBooked} already booked` : null,
+    counts.skipped > 0 ? `${counts.skipped} skipped` : null,
+    counts.other > 0 ? `${counts.other} other` : null,
+  ]
+    .filter(Boolean)
+    .join(' · ');
+
+  const lines = [
+    '## 🏋️ AutoWOD Booking Results',
+    '',
+    '| Day | Time | Status |',
+    '|-----|------|--------|',
+    ...rows,
+    '',
+    totals,
+    '',
+  ];
+
+  appendFileSync(summaryFile, lines.join('\n'));
+}
+
 export async function processReservations(
   page: Page,
   preferences: ReservationPreferences
 ): Promise<void> {
+  const dayResults: Array<{ weekDay: string; result: ReservationResult }> = [];
   let booked = 0;
   let waitlisted = 0;
   let alreadyBooked = 0;
@@ -187,12 +242,13 @@ export async function processReservations(
 
   for (let i = 0; i < availableDays; i++) {
     const weekDay = await getWeekDayFromUrl(page);
-    const time = preferences[weekDay as WeekDay];
+    const preference = preferences[weekDay as WeekDay];
 
-    const result = await makeReservation(page, time);
+    const result = await makeReservation(page, preference);
+    dayResults.push({ weekDay, result });
     console.log(result.message);
 
-    if (!time) {
+    if (!preference) {
       skipped++;
     } else if (result.state === 'Entrenar' && result.success) {
       booked++;
@@ -211,4 +267,6 @@ export async function processReservations(
   console.log(
     `📊 Summary -> booked: ${booked}, waitlist: ${waitlisted}, already booked: ${alreadyBooked}, skipped (no time): ${skipped}, other: ${other}`
   );
+
+  writeJobSummary(dayResults, { booked, waitlisted, alreadyBooked, skipped, other });
 }

